assembl/tasks/piwik.py

```diff
@@ -0,0 +1,321 @@
+
+import string
+import random
+
+import requests
+from zope import interface
+from pyramid.settings import asbool
+
+
+from assembl.lib import config
+from assembl.lib.discussion_creation import IDiscussionCreationCallback
+
+
+class AutomaticPiwikBindingAtDiscussionCreation(object):
+    """A :py:class:`IDiscussionCreationCallback` that creates a Piwik site and user at discussion creation"""
+    interface.implements(IDiscussionCreationCallback)
+
+    def discussionCreated(self, discussion):
+        bind_piwik(discussion)
+
+
+def bind_piwik(discussion, admin=None):
+    admin = admin or discussion.creator
+    assert admin, "There should be a discussion admin provided"
+
+    user_email = admin.get_preferred_email()
+    assert user_email, "the admin should have an email"
+    piwik_url = config.get('web_analytics_piwik_url')
+    piwik_api_token = config.get('web_analytics_piwik_api_token')
+
+    missing_variables = []
+    if not user_email:
+        missing_variables.append("user_email")
+    if not piwik_url:
+        missing_variables.append("piwik_url")
+    if not piwik_api_token:
+        missing_variables.append("piwik_api_token")
+    if len(missing_variables):
+        raise RuntimeError("missing configuration variables: " + ", ".join(missing_variables))
+
+    # TODO: Should this process first check that discussion.web_analytics_piwik_id_site is empty and do something different if it's not? (for example: return an error, so that we empeach discussion statistics to be scattered on different Piwik sites, which is difficult to merge)
+
+    try:
+        # Check wether a Piwik user with a `user_email` email exists
+        try:
+            user_already_exists = piwik_UsersManager_getUserByEmail(piwik_url, piwik_api_token, user_email)
```

Comment:
Ooof. The naming of this function breaks the backend coding style that we've picked. Python naming convention does not include camelCase. Maybe change to `get_user_by_email`?